code/modules/cm_marines/equipment/weapons.dm

```diff
@@ -103,7 +103,43 @@
 		..()
 		if (get_dist(user, src) <= 1)
 			if(pcell)
-				to_chat(user, "A small gauge in the corner reads: Ammo: [rounds_remaining] / 250.")
+				to_chat(user, "A small gauge in the corner reads: Ammo: [rounds_remaining] / [rounds_max].")
+
+/obj/item/smartgun_powerpack/proc/reload(mob/user, obj/item/weapon/gun/smartgun/mygun, automatic = FALSE)
+
+		pcell.charge -= 50
+		if(!mygun.current_mag)
+				var/obj/item/ammo_magazine/internal/smartgun/A = new(mygun)
+				mygun.current_mag = A
+
+		var/rounds_to_reload = min(rounds_remaining, (mygun.current_mag.max_rounds - mygun.current_mag.current_rounds)) //Get the smaller value.
+
+		mygun.current_mag.current_rounds += rounds_to_reload
+		rounds_remaining -= rounds_to_reload
+
+		if(!automatic)	to_chat(user, "You finish loading [rounds_to_reload] shells into the M56 Smartgun. Ready to rumble!")
+		else	to_chat(user, "The powerpack servos finish loading [rounds_to_reload] shells into the M56 Smartgun. Ready to rumble!")
+		playsound(user, 'sound/weapons/unload.ogg', 25, 1)
+
+		reloading = 0
+		return 1
+
+/obj/item/smartgun_powerpack/proc/autoload_check(mob/user, delay, obj/item/weapon/gun/smartgun/mygun, obj/item/smartgun_powerpack/powerpack, numticks = 5)
+	if(!istype(user) || delay <= 0) return FALSE
+
+	var/mob/living/carbon/human/L
+	if(istype(user, /mob/living/carbon/human)) L = user
+
+	var/delayfraction = round(delay/numticks)
+	. = TRUE
+	for(var/i = 0 to numticks)
+		sleep(delayfraction)
+		if(!user)
+			. = FALSE
+			break
+		if(!(L.back == powerpack) || !(L.s_store == mygun) && !(user.get_active_hand() == mygun)) //power pack and gun aren't where they should be.
```

Comment:
This will cause the user to keep reloading without a powerpack as long as the gun remains in the hand.
To accomplish what you want perhaps checking both hands for the gun might be ideal.

Reply:
Yes, you know this because I told you. :p